sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -980,6 +980,13 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
       messageParameters = Map("content" -> content))
   }
 
+  def hintUnsupportedForJdbcDialectError(
+      jdbcDialect: String): SparkUnsupportedOperationException = {
+    new SparkUnsupportedOperationException(
```

Comment:
Please use `SparkIllegalArgumentException` here.
Its an option error.

Reply:
+1 for `SparkIllegalArgumentException` in this context, too.

`hintUnsupportedForJdbcDialectError` method name is proper, but `SparkIllegalArgumentException` would be the correct error message.